### ISSUE TITLE
Default search engine does not change in Private window with tor.

### ISSUE THREAD

**Issue by namelessperson0** [brave/brave-browser#17494]
## Description 
In "New Private Window with Tor", the default search engine is always DuckDuckGo even if I change in the settings. The default search engine does change in Normal window and Private window without tor.

## Steps to Reproduce

   1. Change default search engine to Brave search or Qwant
   2. Go to "New Private Window with Tor"
   3. Search
 

## Actual result:
DuckDuckGo remains the default search engine.

## Expected result:
The search engine mentioned in "Default search engine" in settings should be the search engine in "New Private Window with Tor"
 
## Reproduces how often: 
Always.

## Brave version 
Version 1.28.96 Chromium: 92.0.4515.115 
Linux.

**Comment by Nihal247** [comment]
Hey @namelessperson0, DuckDuckGo is the default one in this case & can't be replaced by any Search Engine in Tor Private Window. This is an expected behaviour.

**Comment by bsclifton** [comment]
cc: @ShivanKaul @rebron

**Comment by catfluoride** [comment]
Is there a possibility for you to reconsider this and allow for custom search engines in TOR windows? Someone (like myself) my want to use one's own SearX instance. If the defaulting to DDG is done for privacy, I guess nothing beats a personal SearX instance in this respect.

Thanks.

**Comment by enc0urage** [comment]
Yes, please. I don't really understand the privacy argument of forcing Brave Search in Tor tabs. Duckduckgo has a Tor hidden service too. There is already a setting to choose the default search engine separately for regular private tabs, why can't there be one for Tor tabs too? Separate option for Tor tabs would be ideal, but it might as well just respect the Private Tab search engine setting.

I don't really have anything against Brave search, I'd like to give it an honest go, but with the hidden service being very slow to rendezvous, I just end up spending up to 30 seconds waiting for the results to load. Often I find faster to hit Ctrl-N and manually type duckduckgo.com and re-type my query. Setting `SingleHop` for the hidden service might help resolve the latency issue. 

Regardless, I'd still like to have the option to choose my own search engine.

**Comment by enc0urage** [comment]
I also noticed that my old installs of Brave still use ddg in Tor tabs, whereas installing Brave on a new machine, I'm now forced to use Brave Search. So is there in fact already a hidden setting/feature flag?
